### PR TITLE
[MLIR] Pass down CMAKE_CXX_FLAGS to external dependency projects

### PR DESCRIPTION
**Pull request by dkalinowski**
Specify _GLIBCXX_USE_CXX11_ABI for protobuf dependency

### CODE REVIEW

File: cmake/external_eigen.cmake [hunk 1]
```diff
@@ -39,6 +39,8 @@ ExternalProject_Add(
     BUILD_COMMAND ""
     INSTALL_COMMAND ""
     EXCLUDE_FROM_ALL TRUE
+    CMAKE_ARGS
```
Comment (silee2) [inline cmake/external_eigen.cmake]: Eigen is header only. You don't need this change since it has no effect

File: cmake/external_json.cmake [hunk 2]
```diff
@@ -35,6 +35,8 @@ ExternalProject_Add(
     INSTALL_COMMAND ""
     UPDATE_COMMAND ""
     EXCLUDE_FROM_ALL TRUE
+    CMAKE_ARGS
```
Comment (silee2) [inline cmake/external_json.cmake]: Json is header only. You don't need this change.

**Comment by silee2** [comment]
nGraph already has a CMake variable for this purpose. Check
https://github.com/NervanaSystems/ngraph/blob/master/CMakeLists.txt#L128-L156

File: cmake/mlir_fetch.cmake.in [hunk 3]
```diff
@@ -32,6 +32,8 @@ ExternalProject_Add(
     SOURCE_DIR ${MLIR_LLVM_ROOT}
     DOWNLOAD_NO_PROGRESS TRUE
     EXCLUDE_FROM_ALL TRUE
+    CMAKE_ARGS
```
Comment (nmostafa) [inline cmake/mlir_fetch.cmake.in]: This has no effect. We just clone LLVM here.

File: cmake/mlir_fetch.cmake.in [hunk 4]
```diff
@@ -44,8 +46,7 @@ ExternalProject_Add(
     CMAKE_GENERATOR ${CMAKE_GENERATOR}
     CMAKE_GENERATOR_PLATFORM ${CMAKE_GENERATOR_PLATFORM}
     CMAKE_GENERATOR_TOOLSET ${CMAKE_GENERATOR_TOOLSET}
-
-    BUILD_COMMAND ${CMAKE_COMMAND} ../llvm -DLLVM_BUILD_EXAMPLES=ON -DLLVM_TARGETS_TO_BUILD=host -DLLVM_ENABLE_RTTI=ON -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+    BUILD_COMMAND ${CMAKE_COMMAND} ../llvm -DLLVM_BUILD_EXAMPLES=ON -DLLVM_TARGETS_TO_BUILD=host -DLLVM_ENABLE_RTTI=ON -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
```
Comment (nmostafa) [inline cmake/mlir_fetch.cmake.in]: Not sure we need this. nGraph C flags wouldn't apply to mlir.

**Pull request closed without merging**